CameraOverlay/commons.py

```diff
@@ -1,47 +1,71 @@
 import os
 import json
+import random
+import fnmatch
 from dotenv import load_dotenv
 
-from .topics import *
-
 load_dotenv()
 CONFIG_FILE = 'configs.json'
 ACTIVE_OVERLAY_KEY = 'activeOverlay'
+OVERLAY_FILE_PATTERN = 'overlay_*.py'
 CURRENT_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
 
 
-def set_overlay(new_overlays, dir=CURRENT_DIRECTORY):
+def get_overlays(directory=CURRENT_DIRECTORY):
+	overlays = [file for file in os.listdir(directory) if fnmatch.fnmatch(file, OVERLAY_FILE_PATTERN)]
+	return overlays
+
+
+def set_overlay(new_overlays, directory=CURRENT_DIRECTORY):
 	current_device = os.getenv('MHP_CAMERA')
 	new_overlay = new_overlays[current_device]
 
-	configs = read_configs(dir)
+	configs = read_configs(directory)
 	if 'device' in configs:
 		configs.pop('device')
 	configs[ACTIVE_OVERLAY_KEY] = new_overlay
 
-	with open(os.path.join(dir, CONFIG_FILE), 'w') as f:
+	with open(os.path.join(directory, CONFIG_FILE), 'w') as f:
 		json.dump(configs, f, indent=2, sort_keys=True)
 
 
-def read_configs(dir=CURRENT_DIRECTORY):
-	filepath = os.path.join(dir, CONFIG_FILE)
-	if not os.path.isfile(filepath):
-		return {}
-	else:
-		with open(filepath) as f:
+def read_configs(directory=CURRENT_DIRECTORY):
+	configs_file = os.path.join(directory, CONFIG_FILE)
+
+	if not os.path.isfile(configs_file):
+		create_default_configs(directory)
+
+	with open(configs_file) as f:
+		configs = json.load(f)
+
+	if ACTIVE_OVERLAY_KEY not in configs:
+		create_default_configs(directory)
+		with open(configs_file) as f:
 			configs = json.load(f)
 
-			current_device = os.getenv('MHP_CAMERA')
-			configs['device'] = current_device
-			return configs
+	current_device = os.getenv('MHP_CAMERA')
+	configs['device'] = current_device
+	configs['overlays'] = get_overlays()
+	return configs
+
+
+def create_default_configs(directory):
+	random_overlay = random.choice(get_overlays(directory))
+	with open(os.path.join(directory, CONFIG_FILE), 'w') as f:
+		json.dump({ACTIVE_OVERLAY_KEY: random_overlay}, f, indent=2, sort_keys=True)
+	return random_overlay
 
 
-def get_active_overlay(dir=CURRENT_DIRECTORY):
-	configs = read_configs(dir)
-	return os.path.join(dir, configs[ACTIVE_OVERLAY_KEY])
+def get_active_overlay(directory=CURRENT_DIRECTORY):
+	configs = read_configs(directory)
+	try:
+		active_overlay = configs[ACTIVE_OVERLAY_KEY]
+	except KeyError:
+		active_overlay = create_default_configs(directory)
+	return os.path.join(directory, active_overlay)
 
 
 if __name__ == '__main__':
-	set_overlay({'primary': 'Overlay_Demo2.py'})
-	print(get_active_overlay())
-	print(read_configs())
+	# set_overlay({'primary': 'overlay_all_stats.py'})
```

Comment:
Why are these comments here?

Reply:
testing purposes, should be fine since it's under the __main__ scope since we're not directly executing the commons.py but using it's functions

Reply:
Given that mentality, it kinda doesn't make sense for this file to be called `commons.py`. Calling it `config.py` would make more sense. So when we import this module to another file, we would run `config.set_overlay(...)`, `config.get_overlays(...)`, `config.read()`, `config.set_default_dir(..)`, `config.get_active_overlay()` etc.

This would make further sense when we run `python3 config.py` and it'll return us the config file! We can later on set arguments to this file so we can set, get overlays manually. `python3 config.py --set "overlay_debug"`. Since this is quite a big change to the file, let's do that in another PR.

Reply:
that's a great idea, i'll put it on trello :)